src/Authentication/Authenticators/Session.php

```diff
@@ -249,28 +285,85 @@ public function loggedIn(): bool
         if ($userId !== null) {
             $this->user = $this->provider->findById($userId);
 
-            return $this->user instanceof User;
+            $identities = $this->userIdentityModel->getIdentitiesByTypes(
+                $this->user->getAuthId(),
+                ['email_2fa', 'email_activate']
```

Comment:
I think you need to get the list of actions assigned in the config file here, otherwise if some wants to do, says, 2FA with SMS messages, this logic would be skipped.

Reply:
I removed the hard coded action types.